Felienne/hedy
# [BUG FIX] Fixes a bug where teachers couldn't view any programs of non-class members

**Description**
This PR should fix the bug where a teacher was unable to view any public program.

**Fixes**
This PR fixes #2174 and fixes #2246

**How to test**
Any user should be able to edit any public program. Regardless of being in a class, being a teacher or any other way around. A public program should always be edit-able. In the case of a teacher also a non-public program of one of their students is edit-able.
